### PR TITLE
gmsh pre-post for FEA

### PR DESCRIPTION
**Pull request by alen12345**
Initial commit, testing the build toolchain, it will take a while...! :)

### CODE REVIEW

**Comment by alen12345** [comment]
Failing at `make install` due to some permission denial.

**Comment by alen12345** [comment]
It would be better to build opencascade before...

File: recipes/gmsh/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}-source.tar.gz
+  url: http://gmsh.info/src/{{ name }}-{{ version }}-source.tgz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: mkdir build && cd build && cmake .. && make && make install
```
Comment (bgruening) [inline recipes/gmsh/meta.yaml]: you probably want to have `cmake -DCMAKE_INSTALL_PREFIX=$PREFIX ..`

**Comment by alen12345** [comment]
@bgruening in my idea, I would build it without OpenCascade support for the moment. Do you agree?

**Comment by bgruening** [comment]
With with me :)

File: recipes/gmsh/meta.yaml (hunk 1 above)
Comment (alen12345) [reply recipes/gmsh/meta.yaml]: Which is the correct install location that is writable?

Comment (alen12345) [reply recipes/gmsh/meta.yaml]: It's just $PREFIX :)

**Comment by alen12345** [comment]
We have two tests failing at the moment.

**Comment by alen12345** [comment]
@CJ-Wright the build is actually massive; travis is failing for an internl server error, but on circleci we are running all tests fine except for opencascade and fltk that we will insert in a future version as agreed also by @bgruening since we need to add those packages too.

Can you please look at the file for the sed part and see it if is ok for you? Thanks!

File: recipes/gmsh/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+source:
+  fn: {{ name }}-{{ version }}-source.tar.gz
+  url: http://gmsh.info/src/{{ name }}-{{ version }}-source.tgz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  # remove the tests of OpenCascade and fltk until we implement them in conda-forge
+  script: sed -i '1919i \ \ list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/tutorial/t16.geo")' CMakeLists.txt && sed -i '1919i \ \ list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/demos/post_processing/lowmem-anim.geo")' CMakeLists.txt && mkdir build && cd build && cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX && make && make test ARGS="-V" && make install
```
Comment (CJ-Wright) [inline recipes/gmsh/meta.yaml]: You might consider formalizing these as a patch https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#patches

Comment (alen12345) [reply recipes/gmsh/meta.yaml]: I must not have understood...! Is for conda a patch file like a bash file?

File: recipes/gmsh/tests.patch [hunk 3]
```diff
@@ -0,0 +1,2 @@
+sed -i '1919i \ \ list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/tutorial/t16.geo")' CMakeLists.txt
```
Comment (CJ-Wright) [inline recipes/gmsh/tests.patch]: I meant a bit more like this:
https://github.com/ilastik/ilastik/wiki/Creating-patches-for-Conda-recipes

Comment (alen12345) [reply recipes/gmsh/tests.patch]: My bad, thought it was a bash script preprocessed by the environment...

Comment (alen12345) [reply recipes/gmsh/tests.patch]: Sorry it is a mistake in my diff, I have done it from a local copy and not with git diff...

**Comment by alen12345** [comment]
@CJ-Wright I am stuck here, if you have time please point me in the right direction. The diff is not applied.

File: recipes/gmsh/tests.patch [hunk 4]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt.edit
+index 5650947..6cdc869 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt.edit
+@@ -1916,6 +1916,8 @@ if(NOT DISABLE_GMSH_TESTS)
+   file(GLOB_RECURSE TESTFILES
+        tutorial/*.geo demos/*.geo benchmarks/?d/*.geo benchmarks/extrude/*.geo
+        benchmarks/boolean/*.geo)
++  list(REMOVE_ITEM TESTFILES "${CMAKE_SOURCE_DIR}/demos/post_processing/lowmem_anim.geo")
```
Comment (CJ-Wright) [inline recipes/gmsh/tests.patch]: This seems inside an if statement, is that correct?

Comment (alen12345) [reply recipes/gmsh/tests.patch]: Yes exactly, it tests if tests are enabled by cmake flags and then builds the list of tests to execute by make test.

**Comment by alen12345** [comment]
@CJ-Wright If you merge fltk from my other pull request I can add it here to make the last failed test pass on linux and mac osx. In the meantime I will try to fix windows too.

**Comment by alen12345** [comment]
@isuruf tests on linux and mac osx are passing apart from one where we need fltk. Windows compilation is crashing; can you please have a look? Is it due to missing c++11 features?

Thanks!

**Pull request closed without merging**